lib/pasarguard-backup.sh

```
@@ -0,0 +1,1314 @@
+            "https://api.telegram.org/bot$BACKUP_TELEGRAM_BOT_KEY/sendDocument" 2>&1)
+
+        local http_code=$(echo "$response" | tail -n1)
+        local response_body=$(echo "$response" | sed '$d')
+
+        if [ "$http_code" == "200" ]; then
+            # Check if response contains "ok":true
+            if echo "$response_body" | grep -q '"ok":true'; then
+                colorized_echo green "Backup part $custom_filename successfully sent to Telegram."
+            else
+                # Extract error message from Telegram response
+                local error_msg=$(echo "$response_body" | grep -o '"description":"[^"]*"' | cut -d'"' -f4 || echo "Unknown error")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)